From: Matt Benjamin <mbenjamin@redhat.com>
Date: Tue, 23 Aug 2016 15:57:06 -0400
Subject: ceph_timer: prefer using directive to typedef

This change replaces the existing member-hook typedef as well as
the new set-type typedefs, so committed separately.

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>
(cherry picked from commit f7d4d418658d2091652033045abdb4bfbeb637aa)

Resolves (with following): rhbz#1370331
---
 src/common/ceph_timer.h | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/common/ceph_timer.h b/src/common/ceph_timer.h
index 97b0dc9..6323be6 100644
--- a/src/common/ceph_timer.h
+++ b/src/common/ceph_timer.h
@@ -58,7 +58,7 @@ namespace ceph {
 
     template <class TC>
     class timer {
-      typedef set_member_hook<link_mode<normal_link> > sh;
+      using sh = set_member_hook<link_mode<normal_link> >;
 
       struct event {
 	typename TC::time_point t;
@@ -89,17 +89,17 @@ namespace ceph {
 	}
       };
 
-      typedef set<event,
-		  member_hook<event, sh, &event::schedule_link>,
-		  constant_time_size<false>,
-		  compare<SchedCompare> > schedule_type;
+      using schedule_type = set<event,
+				member_hook<event, sh, &event::schedule_link>,
+				constant_time_size<false>,
+				compare<SchedCompare> >;
 
       schedule_type schedule;
 
-      typedef set<event,
-		  member_hook<event, sh, &event::event_link>,
-		  constant_time_size<false>,
-		  compare<EventCompare> > event_set_type;
+      using event_set_type = set<event,
+				 member_hook<event, sh, &event::event_link>,
+				 constant_time_size<false>,
+				 compare<EventCompare> >;
 
       event_set_type events;
 
